hospital/apphospital/models.py

```diff
@@ -5,67 +5,83 @@
+        return "ID{} - {}".format(self.pk, self.name)
 
 
 class Department(models.Model):
-    hospital = models.ForeignKey(Hospital, on_delete=models.CASCADE)
-    department_id = models.AutoField(primary_key=True)
-    department_name = models.CharField(max_length=200)
+    hospital = models.ForeignKey(to=Hospital, on_delete=models.CASCADE,
+                                 related_name='departments',
+                                 default='',
+                                 null=False,
+                                 blank=False)
+    name = models.CharField(db_index=True, max_length=200, default='', null=False, blank=False)
 
     def __str__(self):
-        return "ID: {} - department {} in {}".format(self.department_id, self.department_name, self.hospital)
+        return "{}".format(self.name)
 
 
-class People(models.Model):
-    GENDER_TYPES = [('M', 'Male'),
-                    ('F', 'Female'),
-                    ('Other', 'Other')]
-    people_id = models.AutoField(primary_key=True)
-    people_name = models.CharField(max_length=200)
-    age = models.CharField(max_length=200)
-    gender = models.CharField(max_length=200, choices=GENDER_TYPES, default="Male")
+class Person(models.Model):
+    name = models.CharField(db_index=True, max_length=200, default='', null=False, blank=False)
+    age = models.PositiveIntegerField(default=0, null=False, blank=True)
+    gender = models.CharField(max_length=200, choices=[('Male', 'Male'),
+                                                       ('Female', 'Female'),
+                                                       ('Other', 'Other')], null=False, blank=False)
 
     def __str__(self):
-        return self.people_name
-
-
-class Worker(People):
-    JOBS = [('Doc', 'Doctor'),
-            ('Nur', 'Nurse')]
-    role = models.CharField(max_length=50, choices=JOBS, default='Doctor')
-    department = models.ManyToManyField(Department)
+        return self.name
+
+
+class Worker(models.Model):
+    JOBS = [('Doctor', 'Doctor'),
+            ('Nurse', 'Nurse')]
+    Person = models.ForeignKey(to=Person, on_delete=models.CASCADE,
+                               related_name='worker_person',
+                               default='',
+                               null=False,
+                               blank=False)
+    role = models.CharField(max_length=50, choices=JOBS, default='Doctor', null=False, blank=False)
+    department = models.ForeignKey(to=Department, on_delete=models.CASCADE,
+                                   related_name='worker_department',
+                                   default='',
+                                   null=False,
+                                   blank=False)
 
     def __str__(self):
-        return "{} {} works in {}".format(self.role, self.people_name, self.department)
+        return "class {} - {} {} works in {} {}".format(type(self).__name__, self.role, self.Person, self.department,
```

Comment:
Change it to look like this:
"<HospitalWorker {name} working in department {department} in position {position}>"

Reply:
d